SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCleaner.java

```diff
@@ -362,9 +386,19 @@ private void drainTaskCleanupQueue() {
       return;
     }
 
+    Map<String, Integer> incrementalBounceRemainingInstanceMap = new HashMap<>();
```

Comment:
When doing our cleanup, each task doesn't know what has happened to the task before it. This way we can keep track of how many cleaning tasks were already killed in this cleanup run. (i.e. if we have room to kill one cleaning task in our incremental bounce, now we'll know when we get to the second that we've already done that and we shouldn't kill the second)